cd/manager/aws/ecs.go

```diff
@@ -315,77 +327,118 @@ func (e Ecs) updateEcsTaskDefinition(ctx context.Context, taskDefArn, image stri
 		Volumes:                 taskDef.Volumes,
 		Tags:                    []types.Tag{{Key: aws.String(manager.ResourceTag), Value: aws.String(string(e.env))}},
 	}
-	regTaskDefOutput, err := e.ecsClient.RegisterTaskDefinition(ctx, regTaskDefInput)
-	if err != nil {
+	if regTaskDefOutput, err := e.ecsClient.RegisterTaskDefinition(ctx, regTaskDefInput); err != nil {
 		log.Printf("updateEcsTaskDefinition: register task def error: %s, %s, %v", taskDefArn, image, err)
 		return "", err
+	} else {
+		return *regTaskDefOutput.TaskDefinition.TaskDefinitionArn, nil
 	}
-	return *regTaskDefOutput.TaskDefinition.TaskDefinitionArn, nil
 }
 
-func (e Ecs) updateEcsService(cluster, service, image string, tempTask bool) (string, error) {
+func (e Ecs) getEcsTaskDefinition(taskDefArn string) (*types.TaskDefinition, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), manager.DefaultHttpWaitTime)
 	defer cancel()
 
-	// Describe service to get task definition ARN
-	descSvcOutput, err := e.describeEcsService(ctx, cluster, service)
-
-	// Describe task to get full task definition
-	newTaskDefArn, err := e.updateEcsTaskDefinition(ctx, *descSvcOutput.Services[0].TaskDefinition, image)
-	family := e.taskFamilyFromArn(newTaskDefArn)
+	input := &ecs.DescribeTaskDefinitionInput{
+		TaskDefinition: aws.String(taskDefArn),
+	}
+	if output, err := e.ecsClient.DescribeTaskDefinition(ctx, input); err != nil {
+		log.Printf("getEcsTaskDefinition: describe task def error: %s, %v", taskDefArn, err)
+		return nil, err
+	} else {
+		return output.TaskDefinition, nil
+	}
+}
 
+func (e Ecs) updateEcsService(cluster, service, image string, tempTask bool) (string, error) {
+	// Describe service to get task definition ARN
+	descSvcOutput, err := e.describeEcsService(cluster, service)
+	if err != nil {
+		log.Printf("updateEcsService: describe service error: %s, %s, %s, %v, %v", cluster, service, image, tempTask, err)
+		return "", err
+	}
+	// Update task definition with new image
+	newTaskDefArn, err := e.updateEcsTaskDefinition(*descSvcOutput.Services[0].TaskDefinition, image)
+	if err != nil {
+		log.Printf("updateEcsService: update task def error: %s, %s, %s, %v, %v", cluster, service, image, tempTask, err)
+		return "", err
+	}
+	// For "temporary" tasks, i.e. tasks that come up, do their work, then exit, the number of running tasks should be
+	// set to 0.
+	var desiredCount int32 = 1
```

Comment:
Please check this section, @v-stickykeys. I don't think I should do this with CASv1 because it will override the number of desired tasks. If anything, I think I can leave this setting entirely unspecified (even for CASv2) so that it just comes from the ECS Service settings.

Reply:
And if it's a mandatory parameter, I'm already describing the service at the start of the function, and can just copy over the previous value to the update.